presto-main/src/test/java/com/facebook/presto/operator/TestHashJoinOperator.java

```diff
@@ -326,26 +400,68 @@ public void testInnerJoinWithSpill(boolean probeHashEnabled, List<WhenSpill> whe
     }
 
+    private enum WhenSpillFails
+    {
+        SPILL_HBO, SPILL_LJO, UNSPILL_HBO, UNSPILL_LJO
+    }
+
     @DataProvider
     public Object[][] testInnerJoinWithSpillDataProvider()
     {
-        List<Object[]> result = new ArrayList<>();
+        return joinWithSpillParameters(true).stream()
+                .map(List::toArray)
+                .toArray(Object[][]::new);
+    }
+
+    private static List<List<Object>> joinWithSpillParameters(boolean allowNoSpill)
+    {
+        List<List<Object>> result = new ArrayList<>();
         for (boolean probeHashEnabled : ImmutableList.of(false, true)) {
             for (WhenSpill whenSpill : WhenSpill.values()) {
                 // spill all
-                result.add(new Object[] {probeHashEnabled, nCopies(PARTITION_COUNT, whenSpill)});
+                if (allowNoSpill || whenSpill != WhenSpill.NEVER) {
+                    result.add(ImmutableList.of(probeHashEnabled, nCopies(PARTITION_COUNT, whenSpill)));
+                }
 
                 if (whenSpill != WhenSpill.NEVER) {
                     // spill one
-                    result.add(new Object[] {probeHashEnabled, concat(singletonList(whenSpill), nCopies(PARTITION_COUNT - 1, WhenSpill.NEVER))});
+                    result.add(ImmutableList.of(probeHashEnabled, concat(singletonList(whenSpill), nCopies(PARTITION_COUNT - 1, WhenSpill.NEVER))));
                 }
             }
 
-            result.add(new Object[] {probeHashEnabled, concat(asList(WhenSpill.DURING_BUILD, WhenSpill.AFTER_BUILD), nCopies(PARTITION_COUNT - 2, WhenSpill.NEVER))});
-            result.add(new Object[] {probeHashEnabled, concat(asList(WhenSpill.DURING_BUILD, WhenSpill.DURING_USAGE), nCopies(PARTITION_COUNT - 2, WhenSpill.NEVER))});
+            result.add(ImmutableList.of(probeHashEnabled, concat(asList(WhenSpill.DURING_BUILD, WhenSpill.AFTER_BUILD), nCopies(PARTITION_COUNT - 2, WhenSpill.NEVER))));
+            result.add(ImmutableList.of(probeHashEnabled, concat(asList(WhenSpill.DURING_BUILD, WhenSpill.DURING_USAGE), nCopies(PARTITION_COUNT - 2, WhenSpill.NEVER))));
         }
+        return result;
+    }
+
+    @DataProvider
+    public Object[][] testInnerJoinWithFailingSpillDataProvider()
+    {
+        List<List<Object>> spillFailValues = Arrays.stream(WhenSpillFails.values())
+                .map(ImmutableList::<Object>of)
+                .collect(toList());
+        return product(joinWithSpillParameters(false), spillFailValues).stream()
+                .map(List::toArray)
+                .toArray(Object[][]::new);
+    }
+
+    private static void processRow(final Driver ljoDriver, final TaskStateMachine taskStateMachine)
+            throws Exception
+    {
+        ljoDriver.getDriverContext().getYieldSignal().setWithDelay(TimeUnit.SECONDS.toNanos(1), ljoDriver.getDriverContext().getYieldExecutor());
+        ljoDriver.process();
+        ljoDriver.getDriverContext().getYieldSignal().reset();
```

Comment:
This should be in a finally:
```java 
ljoDriver.getDriverContext().getYieldSignal().setWithDelay(TimeUnit.SECONDS.toNanos(1), ljoDriver.getDriverContext().getYieldExecutor());
try {
        ljoDriver.process();
} finally {
        ljoDriver.getDriverContext().getYieldSignal().reset();
}

Reply:
I think it's unnecessary. When `ljoDriver.process()` throws, we're not interested in yield signal anymore.

Reply:
I asked the author of the yield code and he said it must be written this way.

Reply:
hm.... maybe there is some misunderstanding.. since the very `com.facebook.presto.operator.TestHashJoinOperator#testYield` test doesn't use finally. Let's not do this here.